codegen-core/src/main/kotlin/software/amazon/smithy/rust/codegen/core/smithy/protocols/serialize/EventStreamMarshallerGenerator.kt

```diff
@@ -109,6 +109,25 @@ open class EventStreamMarshallerGenerator(
         }
     }
 
+    fun renderInitialResponseGenerator(contentType: String): RuntimeType {
```

Comment:
1. Isn't this functionality server-specific? If so, shouldn't we move it out of the core module and keep it only in server-related modules?

2. Don't REST-based protocols use headers only, while RPC protocols use the body? If that's the case, wouldn't this code be inapplicable for RPC-based protocols?

Reply:
you _could_ do that for ideological reasons, but all the library functions it needs are in here so its cleaner IMO